spicy/toolchain/bin/spicy-dump/printer-json.cc

```
@@ -13,130 +15,139 @@ using json = nlohmann::json;
+                    auto o = json::object();
+
+                    if ( offset ) {
+                        o["start"] = std::get<0>(*offset).Ref();
+                        if ( auto& end = std::get<1>(*offset) )
+                            o["end"] = end->Ref();
+                    }
+
+                    offsets.push_back(std::move(o));
+                }
+                j["__offsets"] = offsets;
+            }
+
+            return j;
+        }
+        case TypeInfo::Time: return type.time->get(v).seconds();
+        case TypeInfo::Tuple: {
+            auto j = json::array();
+
+            for ( auto i : type.tuple->iterate(v) )
+                j.push_back(convert(i.second));
+
+            return j;
+        }
+        case TypeInfo::Union: {
+            auto y = type.union_->value(v);
+            return y ? convert(y) : json();
+        }
+        case TypeInfo::UnsignedInteger_uint8_t: return type.unsigned_integer_uint8->get(v);
+        case TypeInfo::UnsignedInteger_uint16_t: return type.unsigned_integer_uint16->get(v);
+        case TypeInfo::UnsignedInteger_uint32_t: return type.unsigned_integer_uint32->get(v);
+        case TypeInfo::UnsignedInteger_uint64_t: return type.unsigned_integer_uint64->get(v);
+        case TypeInfo::ValueReference: {
+            auto y = type.value_reference->value(v);
+            return y ? convert(y) : json();
+        }
+        case TypeInfo::Vector: {
+            auto j = json::array();
+
+            for ( auto i : type.vector->iterate(v) )
+                j.push_back(convert(i));
+
+            return j;
+        }
+        case TypeInfo::VectorIterator: return convert(type.vector_iterator->value(v));
+        case TypeInfo::Void: return "<void>";
+        case TypeInfo::WeakReference: {
+            auto y = type.weak_reference->value(v);
+            return y ? convert(y) : json();
+        }
+    }
+
+    throw RuntimeError("unhandled type");
```

move into `default` case? the compiler will probably not let this get executed anyways, but instead complain about a case not handled if added more type tag.

---

Ideally I wouldn't have added this potentially dead code, but have relied on the compiler checking that the cases are exhaustive. I added it here since I believe GCC could warn that such a function wouldn't return a value.

I'd suggest don't use a `default` since it would mask the warning about an unhandled case.